base/common/src/main/java/com/netscape/certsrv/profile/PolicyConstraintValue.java

```diff
@@ -120,18 +121,119 @@ public boolean equals(Object obj) {
+            }
+
+            if (descriptor.getDefaultValue() != null) {
+                Element defaultValueElement = document.createElement("mDef");
+                defaultValueElement.appendChild(document.createTextNode(descriptor.getDefaultValue()));
+                descriptorElement.appendChild(defaultValueElement);
+            }
+            pcvElement.appendChild(descriptorElement);
+        }
+
+        if (value != null) {
+            Element valueElement = document.createElement("value");
+            valueElement.appendChild(document.createTextNode(value.toString()));
+            pcvElement.appendChild(valueElement);
+        }
+
+        return pcvElement;
+    }
+
+    public static PolicyConstraintValue fromDOM(Element pcvElement) {
+
+        PolicyConstraintValue pcv = new PolicyConstraintValue();
+
+        String id = pcvElement.getAttribute("id");
+        pcv.setName(id);
+
+        NodeList valueList = pcvElement.getElementsByTagName("value");
+        if (valueList.getLength() > 0) {
+            String value = valueList.item(0).getTextContent();
+            pcv.setValue(value);
+        }
+
+        NodeList descriptorList = pcvElement.getElementsByTagName("descriptor");
+        if (descriptorList.getLength() > 0) {
+            String syntax = null;
+            String constraint = null;
+            String description = null;
+            String def = null;
+            NodeList syntaxList = pcvElement.getElementsByTagName("mSyntax");
+            NodeList constraintList = pcvElement.getElementsByTagName("mConstraint");
+            NodeList descriptionList = pcvElement.getElementsByTagName("mDescription");
+            NodeList defList = pcvElement.getElementsByTagName("mDef");
+            if (syntaxList.getLength() > 0) {
+                syntax = syntaxList.item(0).getTextContent();
+            }
+            if (constraintList.getLength() > 0) {
+                constraint = constraintList.item(0).getTextContent();
+            }
+            if (descriptionList.getLength() > 0) {
+                description = descriptionList.item(0).getTextContent();
+            }
+            if (defList.getLength() > 0) {
+                def = defList.item(0).getTextContent();
+            }
+            Descriptor descriptor = new Descriptor(syntax, constraint, def, description);
+            pcv.setDescriptor(descriptor);
+        }
+
+        return pcv;
+    }
+
     public String toXML() throws Exception {
-        Marshaller marshaller = JAXBContext.newInstance(PolicyConstraintValue.class).createMarshaller();
-        marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, true);
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+        Document document = builder.newDocument();
+
+        Element accountElement = toDOM(document);
+        document.appendChild(accountElement);
```

Comment:
We probably should call it `rootElement` or just `element`.

Reply:
Renamed

Reply:
As above